client/client.go

```diff
@@ -923,16 +926,28 @@ func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
 	}
 
 	var (
-		pubKey    data.PublicKey
-		err       error
-		errFmtMsg string
+		pubKey     data.PublicKey
+		pubKeyList data.KeyList
+		err        error
+		errFmtMsg  string
 	)
-	switch serverManagesKey {
-	case true:
+
+	switch {
```

Comment:
Currently keys are checked to be "root keys" by checking that they're encrypted. If we want to support specifying keys for all rotations, I can amend this PR to include that, but it's not the current behavior.